libminifi/src/utils/ListingStateUtils.cpp

```diff
@@ -0,0 +1,95 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "utils/ListingStateUtils.h"
+
+#include "core/Property.h"
+
+namespace org::apache::nifi::minifi::utils {
+
+const std::string ListingStateManager::LATEST_LISTED_OBJECT_PREFIX = "listed_key.";
+const std::string ListingStateManager::LATEST_LISTED_OBJECT_TIMESTAMP = "listed_timestamp";
+
+bool ListingState::wasObjectListedAlready(const ListedObject &object) const {
+  return listed_key_timestamp > object.getLastModified() ||
+      (listed_key_timestamp == object.getLastModified() && listed_keys.find(object.getKey()) != listed_keys.end());
+}
+
+void ListingState::updateState(const ListedObject &object) {
+  if (listed_key_timestamp < object.getLastModified()) {
+    listed_key_timestamp = object.getLastModified();
+    listed_keys.clear();
+    listed_keys.insert(object.getKey());
+  } else if (listed_key_timestamp == object.getLastModified()) {
+    listed_keys.insert(object.getKey());
+  }
+}
+
+uint64_t ListingStateManager::getLatestListedKeyTimestamp(const std::unordered_map<std::string, std::string> &state) const {
```

Comment:
Clang-Tidy: Method 'getLatestListedKeyTimestamp' can be made static

Reply:
Updated in ef0172d119c472a1c91a10aecbe531a112ade2bf